### ISSUE TITLE
Allow viewing the edit history of comments

### ISSUE THREAD

**Issue by aensley** [dear-github/dear-github#129]
Related to #128. Often, comments need to be edited, and that's a great feature. However, sometimes, comments are edited for not-so-good reasons, and there's no way to see the original, when it was edited, or by whom.

The GitHub API seems to offer all versions of the comments, but that information is hidden in GitHub's interface.

**Comment by vadmeste** [comment]
:+1:

**Comment by codynguyen1116** [comment]
+1

**Comment by koenpunt** [comment]
:+1: Would really like history on comments and issues. 

Had a guy promoting his repo by e-mail, saying he figured I would be interested in some random project.
I opened an issue on this repo to make my annoyance clear, and after a short conversation, he removed by comments, updated my original issue body, and updated the title to `[Invalid ticket]`

**Comment by joshmanders** [comment]
@koenpunt Yup, been noticing same issues with other malicious users. Deleting issue comments and retitling to `Undefined` or something.

**Comment by hagronnestad** [comment]
:+1:

**Comment by WooGenius** [comment]
:+1:

**Comment by ElvisAttro** [comment]
+1

**Comment by vinicius0026** [comment]
+1
By the way, is it possible to access this information via API? I was not able to do so... Any help is appreciated.

**Comment by aensley** [comment]
You can see when the issue was created and updated, but it still only gives you the last version of the body:

https://api.github.com/repos/dear-github/dear-github/issues/129

```JSON
{
  "url": "https://api.github.com/repos/dear-github/dear-github/issues/129",
  "repository_url": "https://api.github.com/repos/dear-github/dear-github",
  "labels_url": "https://api.github.com/repos/dear-github/dear-github/issues/129/labels{/name}",
  "comments_url": "https://api.github.com/repos/dear-github/dear-github/issues/129/comments",
  "events_url": "https://api.github.com/repos/dear-github/dear-github/issues/129/events",
  "html_url": "https://github.com/dear-github/dear-github/issues/129",
  "id": 134635222,
  "number": 129,
  "title": "Allow viewing the edit history of comments",
  "user": {
    "login": "aensley",
    "id": 95717,
    "avatar_url": "https://avatars.githubusercontent.com/u/95717?v=3",
    "gravatar_id": "",
    "url": "https://api.github.com/users/aensley",
    "html_url": "https://github.com/aensley",
    "followers_url": "https://api.github.com/users/aensley/followers",
    "following_url": "https://api.github.com/users/aensley/following{/other_user}",
    "gists_url": "https://api.github.com/users/aensley/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/aensley/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/aensley/subscriptions",
    "organizations_url": "https://api.github.com/users/aensley/orgs",
    "repos_url": "https://api.github.com/users/aensley/repos",
    "events_url": "https://api.github.com/users/aensley/events{/privacy}",
    "received_events_url": "https://api.github.com/users/aensley/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "open",
  "locked": false,
  "assignee": null,
  "milestone": null,
  "comments": 8,
  "created_at": "2016-02-18T16:54:31Z",
  "updated_at": "2016-04-11T21:36:10Z",
  "closed_at": null,
  "body": "Related to #128. Often, comments need to be edited, and that's a great feature. However, sometimes, comments are edited for not-so-good reasons, and there's no way to see the original, when it was edited, or by whom.\r\n\r\nThe GitHub API seems to offer all versions of the comments, but that information is hidden in GitHub's web interface.\r\n\r\n**Edit**\r\nAdding content here in an edit to see if/when the history shows up.",
  "closed_by": null
}
```

**Comment by aensley** [comment]
Same thing with comments:

https://api.github.com/repos/dear-github/dear-github/issues/comments/208573845

And now to copy in the escaped JSON content inside a JSON response so that GitHub's API/web interface can really freak out.

```JSON
{
  "url": "https://api.github.com/repos/dear-github/dear-github/issues/comments/208573845",
  "html_url": "https://github.com/dear-github/dear-github/issues/129#issuecomment-208573845",
  "issue_url": "https://api.github.com/repos/dear-github/dear-github/issues/129",
  "id": 208573845,
  "user": {
    "login": "aensley",
    "id": 95717,
    "avatar_url": "https://avatars.githubusercontent.com/u/95717?v=3",
    "gravatar_id": "",
    "url": "https://api.github.com/users/aensley",
    "html_url": "https://github.com/aensley",
    "followers_url": "https://api.github.com/users/aensley/followers",
    "following_url": "https://api.github.com/users/aensley/following{/other_user}",
    "gists_url": "https://api.github.com/users/aensley/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/aensley/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/aensley/subscriptions",
    "organizations_url": "https://api.github.com/users/aensley/orgs",
    "repos_url": "https://api.github.com/users/aensley/repos",
    "events_url": "https://api.github.com/users/aensley/events{/privacy}",
    "received_events_url": "https://api.github.com/users/aensley/received_events",
    "type": "User",
    "site_admin": false
  },
  "created_at": "2016-04-11T21:38:47Z",
  "updated_at": "2016-04-11T21:39:02Z",
  "body": "You can see when the issue was created and updated, but it still only gives you the last version of the body:\r\n\r\n\r\nhttps://api.github.com/repos/dear-github/dear-github/issues/129\r\n\r\n\r\n```JSON\r\n{\r\n  \"url\": \"https://api.github.com/repos/dear-github/dear-github/issues/129\",\r\n  \"repository_url\": \"https://api.github.com/repos/dear-github/dear-github\",\r\n  \"labels_url\": \"https://api.github.com/repos/dear-github/dear-github/issues/129/labels{/name}\",\r\n  \"comments_url\": \"https://api.github.com/repos/dear-github/dear-github/issues/129/comments\",\r\n  \"events_url\": \"https://api.github.com/repos/dear-github/dear-github/issues/129/events\",\r\n  \"html_url\": \"https://github.com/dear-github/dear-github/issues/129\",\r\n  \"id\": 134635222,\r\n  \"number\": 129,\r\n  \"title\": \"Allow viewing the edit history of comments\",\r\n  \"user\": {\r\n    \"login\": \"aensley\",\r\n    \"id\": 95717,\r\n    \"avatar_url\": \"https://avatars.githubusercontent.com/u/95717?v=3\",\r\n    \"gravatar_id\": \"\",\r\n    \"url\": \"https://api.github.com/users/aensley\",\r\n    \"html_url\": \"https://github.com/aensley\",\r\n    \"followers_url\": \"https://api.github.com/users/aensley/followers\",\r\n    \"following_url\": \"https://api.github.com/users/aensley/following{/other_user}\",\r\n    \"gists_url\": \"https://api.github.com/users/aensley/gists{/gist_id}\",\r\n    \"starred_url\": \"https://api.github.com/users/aensley/starred{/owner}{/repo}\",\r\n    \"subscriptions_url\": \"https://api.github.com/users/aensley/subscriptions\",\r\n    \"organizations_url\": \"https://api.github.com/users/aensley/orgs\",\r\n    \"repos_url\": \"https://api.github.com/users/aensley/repos\",\r\n    \"events_url\": \"https://api.github.com/users/aensley/events{/privacy}\",\r\n    \"received_events_url\": \"https://api.github.com/users/aensley/received_events\",\r\n    \"type\": \"User\",\r\n    \"site_admin\": false\r\n  },\r\n  \"labels\": [\r\n\r\n  ],\r\n  \"state\": \"open\",\r\n  \"locked\": false,\r\n  \"assignee\": null,\r\n  \"milestone\": null,\r\n  \"comments\": 8,\r\n  \"created_at\": \"2016-02-18T16:54:31Z\",\r\n  \"updated_at\": \"2016-04-11T21:36:10Z\",\r\n  \"closed_at\": null,\r\n  \"body\": \"Related to #128. Often, comments need to be edited, and that's a great feature. However, sometimes, comments are edited for not-so-good reasons, and there's no way to see the original, when it was edited, or by whom.\\r\\n\\r\\nThe GitHub API seems to offer all versions of the comments, but that information is hidden in GitHub's web interface.\\r\\n\\r\\n**Edit**\\r\\nAdding content here in an edit to see if/when the history shows up.\",\r\n  \"closed_by\": null\r\n}\r\n```"
}
```

**Comment by vinicius0026** [comment]
Yeah, I got that... The edit history of the issue description is what I was looking for. Thanks anyway!

**Comment by aensley** [comment]
@vinicius0026 Right. And that's what this issue is asking for. :-)

**Comment by Qix-** [comment]
Github now shows when comments are edited, and who by - but not the edit history.

I'm actually in favor of this - if someone posts sensitive information in a comment and then redacts it later (or the company redacts it for them, or a bot, etc.) then it shouldn't be there to show in the history as a safety consideration.

I'd consider this closed to be fair. The only improvement they could make (and I haven't seen whether or not this is already done) is to just show *by whom* and *when*, in a list, it was edited.

**Comment by FichteFoll** [comment]
Regarding edit history, there has recently been an update blog post about integrations and commit/issue editing. Specifically, you get to see the old and the new content of edited comments (and issues) in webhooks, so technically one could develop such an integration externally. However, I have the feeling that it won't be too long until github adds this feature itself. :+1: 

https://github.com/blog/2156-expanded-webhook-events

**Comment by muesliq** [comment]
+1

- [ ] Nobody can
- [ ] find out
- [ ] who checked these tasks 
- [ ] (and when)

**Comment by RobertBColton** [comment]
I think what would be an acceptable solution is to allow only users to see their own edit history and nobody else be able to see them. Another possibility is to make a setting for each repo the same way public and private repos are designated.

**Comment by jesus2099** [comment]
I think seeing our own edits is almost no use, except your particular example.
The main interest of seeing edit history is by others, it allows to better understand the replies to a later edited post, for instance.

**Comment by FichteFoll** [comment]
Discourse handles editing really well, although it has a time limit on how long you can edit your posts which github currently does not. As long as OPs remain editable, I'm find with either.

**Comment by dnesting** [comment]
If GitHub implements this it is necessary for them to also have a method by which we can permanently delete information, lest PII or other sensitive data accidentally make its way into a comment.  Also do not enable this feature retroactively.

**Comment by aensley** [comment]
@dnesting It may be impossible to implement retroactively anyway. I don't know if they're saving the original comments, but I doubt it. As for sensitive information, I think the maxim "The Internet is forever" is appropriate.

If somebody posts something on a public project page, they should have no expectations of being able to delete it, even if that feature is available.

**Comment by dnesting** [comment]
1. Not all comments are on publicly-visible repos
2. Just because something could have been observed doesn't mean that it is impossible for further harm to occur by continuing to make it available.  People should mitigate risk as if the information were exposed to people eager to exploit it, but that doesn't mean we shouldn't have tools that let us remove the information from view, to reduce the potential for that information to be exploited by others.  It's easy to change passwords and secret keys, but not so easy to reduce the harm caused by exposed health records, for instance.

**Comment by aensley** [comment]
I work in several private repos myself. I assumed you were talking about information being exposed publicly by not being able to delete it. My mistake.

**Comment by jhawlwut** [comment]
Here's a scenario regarding this topic: code reviews to meet compliance standards.

So, if you have a code review process utilizing PRs to the master branch of a private repository, it'd be great if the Org/Repo owner(s) or admin could lock down comment editing. This assumes a few things: 
- If a collaborator is commenting in a private repo they are a member of that organization and or have been added as a collaborator
- If sensitive information has been posted, it's up to the owners / admins to remove the post

A big challenge right now is if an approved says a review looks good, only to edit that comment later. A workaround is labels, but verifying the original comment would be great.

Sorry for any misspellings, iPhone typed.

**Comment by aensley** [comment]
@jhawlwut That's exactly the atmosphere I'm coming from, but I know any solution/change has to work for everyone.

**Comment by cvvergara** [comment]
I ask for another scenario:
What happens when people on good faith posted a code snipet, and it happened that later he realizes that by posting the snipet he is breaking some license/agreement, and wants to delete it, including from the history?

**Comment by aensley** [comment]
@cvvergara I think that would be covered by my suggestion above: https://github.com/dear-github/dear-github/issues/129#issuecomment-224027073

**Comment by flowmemo** [comment]
+1
BTW, is there a way to get updated time of the "0th" comment(issue's description), e.g. https://github.com/dear-github/dear-github/issues/129#issue-134635222 ?  The API for issue's comments is not usable.

**Comment by kiramclean** [comment]
+1

**Comment by mattbrunetti** [comment]
+1

**Comment by mattbrunetti** [comment]
Is there any third party integration yet that makes use of the web hook and collects a history of comment edits?

**Comment by mtaufen** [comment]
+∞

**Comment by mattbrunetti** [comment]
This is a serious problem when a developer is forced to work with a vendor that actively deceives and confuses issues...

**Comment by sergeevabc** [comment]
Since English is not my native language, I frequently edit my comments within the frame of a few minutes to make point more precise. Gitter keeps number of history of edits and edits itself for others to see and it makes me feel uncomfortable, whereas Github saves the last edit only and I’m happy with it, that’s why I downvote this request.

**Comment by dylantayloribm** [comment]
IBM, my employer, is using Github Enterprise, and this is a feature that would be very useful to us. Not being able to tell who edits a comment or what was edited is definitely not ideal. In the old solution our team used, RTC, we could see diffs for changes on work items and comments on these items.

**Comment by mattbrunetti** [comment]
@sergeevabc Thanks for the info. For my own personal interests I'm on the "let me see the history" camp, but it's critical that we know all the reasons why or why not. 

Even with English being my native language, I find it extremely helpful to frequently edit the comment for the next few minutes to make my point more clear and simple. It's for that reason I want to be able to see the history; Because we can't see the history, and the vendor we were working with would deliberately try to deceive and confuse, we had to make a strict policy of not editing comments and instead just adding a new comment. And so I completely lost that ability to edit/refine/correct my comments. The root problem here is being forced to work with an evil vendor (that problem is fixed now btw) but being able to see the history would have mitigated that problem.

Why would other people seeing the # and content of your edits make you uncomfortable? The way I see it there is no shame in English not being your native language. BTW your English seems excellent to me, at least in the last version of your comment ;)

**Comment by FichteFoll** [comment]
@sergeevabc, @mattbrunetti, you should be interested in the comment I made earlier: https://github.com/dear-github/dear-github/issues/129#issuecomment-222392638

**Comment by bdjnk** [comment]
Another argument in favor of full comment history, including content, being accessible, is the "solved" problem. Someone will ask a question or suggest something might be a bug, then, once they are provided with or discover the answer, will edit their original post to say "solved" or "nevermind" or some such nonsense. Though this seems a product of thoughtlessness and inexperience, not malicious intent, it is nevertheless obnoxious.

Regarding sensitive data redaction, although I think it somewhat a red herring, perhaps give the repository owner authority to delete segments which would then show up as [REDACTED] in the content history view.

**Comment by Qix-** [comment]
@bdjnk The first case isn't common enough I don't think, and your second case would have clunky or magic UX unfortunately.

**Comment by aensley** [comment]
Fear-mongering with WannaCry over comment edit history. lol.

No one is suggesting not having any way to permanently delete information, especially for very sensitive (and extremely rare) comments.

A solution for those instances was suggested above: https://github.com/dear-github/dear-github/issues/129#issuecomment-224027073

**Comment by Qix-** [comment]
It's not fear mongering. Most repositories with heavy security implications have [responsible disclosure processes](https://www.openssl.org/news/vulnerabilities.html). I _guarantee_ you they do not involve including information in a Github issue.

**Comment by roberthopman** [comment]
When is this going to be implemented?

**Comment by sergeevabc** [comment]
> When is this going to be implemented?

Never. Public user has a right to be forgotten, partially (last edit is presented only) or completely.

**Comment by aensley** [comment]
Absolute garbage. The Internet is forever. Anyone who thinks otherwise is lying to themselves.

For the last time: No one is suggesting not having any way to permanently delete information, especially for very sensitive (and extremely rare) comments.

A solution for those instances was suggested above: https://github.com/dear-github/dear-github/issues/129#issuecomment-224027073

**Comment by jhawlwut** [comment]
This really goes back to organizational-level control. If you work for a company with strict security guidelines or compliance-driven development, the organization should have fine grain ability to control the history of a repository. This includes control of user comment(s).

**Comment by Qix-** [comment]
> strict security guidelines or compliance-driven development

Those companies wouldn't use public Github, either.

**Comment by jhawlwut** [comment]
@Qix- having come from one, I respectfully disagree.

**Comment by aensley** [comment]
@jhawlwut And that is _accounted for_ in the proposed changes.

**Comment by Qix-** [comment]
You have code that has strict security guidelines (e.g. trade secrets, encryption standards, etc) and you're hosting your code in what is essentially _'the cloud'_?

Having come from one myself, it's a bit of a terrifying thing to thing about getting audited and having to explain why potentially sensitive code is being paraded around servers that we don't control.

**Comment by SteveJobzniak** [comment]
We often have to edit user comments because they publicly post their own passwords, security tokens etc when posting debug logs.

Please do not disable our ability to protect them from their own stupidity. 🤣 

Or perhaps we should let nature take its course...

**Comment by aensley** [comment]
_Replying to https://github.com/dear-github/dear-github/issues/129#issuecomment-333705880 by @SteveJobzniak:_

https://github.com/dear-github/dear-github/issues/129#issuecomment-224027073

**Comment by SteveJobzniak** [comment]
@aensley +1 to what you said in that comment. But you seem to suggest the current Github system. You say allowing editing by repository team members and the original author. That's the way it is today.

The only difference I see is that you think there should be a public log of each edit, but not what content changed. Hmm. Why? What help does the log of "posted by X, edited by Y, edited by Y" provide? Compared to Github's current "posted by X, edited by Y".

**Comment by aensley** [comment]
_Replying to https://github.com/dear-github/dear-github/issues/129#issuecomment-333888465 by @SteveJobzniak:_

Yes, that is exactly what this issue, titled "Allow viewing the edit history of comments" is for.

My suggested solution in that link above allows for viewing the edited content, time, and author for each comment _unless_ something really needs to be deleted like you mentioned in your comment. In that case, the author and time would still be retained, but not the content. I strongly believe there needs to be the ability to permanently delete content for the many reasons stated above by others, but that it should not be the default. In my experience, people aren't editing comments to remove sensitive information most of the time.

**Comment by SteveJobzniak** [comment]
@aensley I like the idea then. Except that it would make life hard for us since our users post their tokens along with their questions and debug logs. So we edit it to remove the tokens so that nobody steals their accounts.

With your suggestion, Charles Darwin would have his day. ;-)

**Comment by aensley** [comment]
_Replying to https://github.com/dear-github/dear-github/issues/129#issuecomment-334006420 by @SteveJobzniak:_

Ah, I see the problem, and this explains the previous misunderstanding. Let me clarify these parts:

> What if the ability to permanently delete content was limited to the comment author and repository owners only?

> 2. Still allowing the ability to permanently delete accidentally-posted sensitive information by those who should have that ability/responsibility.

What I meant was allowing the ability to remove only the content of _past revisions_ of the comment. Entirely deleting a comment is and will always be possible.

I'm envisioning an edit history similar to what GitHub Wiki gives us:

![wiki_edits](https://user-images.githubusercontent.com/95717/31196128-3d304dfa-a912-11e7-949e-1b00271ae1f9.png)

Then you could compare revisions, and optionally delete the content of some revisions (to remove sensitive information).

Sorry for my previous snark. Was having a bad day.

**Comment by SteveJobzniak** [comment]
OH... @aensley that is a brilliant idea! I love it. So then we could review history, and delete specific revisions if necessary, or delete the whole comment if necessary. That's great and solves everything.

And no offense taken. I get bad days too and fully understand. :-) Hope you rest and relax! :-D

**Comment by benbucksch** [comment]
[X] We had a list of tasks
[X] I was adding a lot of tasks, merged from other comments into the first description, and I deleted the comments
[X] I reloaded and my new description was good
[X] Somebody else checked a number of tasks
[ ] His edit was based on the older version of the description
[ ] His edit overwrote my changes
[ ] There is no way for me to get the moved tasks back.
[ ] The comments are deleted permanently
[ ] The new description is also overwritten and lost

-> DATALOSS

**Comment by SteveJobzniak** [comment]
@benbucksch Uhm... ouch, but uhm... that's not what tickets are for. That's what GitHub Projects are for:

https://github.com/benbucksch/corvette/projects

https://github.com/benbucksch/jscollection/projects

https://github.com/benbucksch/trex/projects

**Comment by benbucksch** [comment]
FWIW, we never intended to collaboratively edit. It was a one-time accident. That's not the point, though. There are many other ways to run into the same problem.

The point is that old versions of comments are lost. And even other people can edit my comments. That's dataloss.

**Comment by SteveJobzniak** [comment]
@benbucksch Agreed.

**Comment by FichteFoll** [comment]
You can now see how many times a comment has been edited, when and by whom via a dropdown. That's quite an improvement, imho, although you don't see what was changed now.

**Comment by sergeevabc** [comment]
@FichteFoll, another data mining monster is rising.

**Comment by aensley** [comment]
_Replying to https://github.com/dear-github/dear-github/issues/129#issuecomment-342576765 by @FichteFoll:_

That is awesome! Very big progress!

**Comment by hoganlong** [comment]
This is a version control system -- why not have version control?  It is a mystery.

**Comment by Qix-** [comment]
IMO this new feature is as far as Github should take it.

**Comment by GolfingSuccess** [comment]
I believe everybody should be able to view a comment's edit history, I don't see why they shouldn't be.

**Comment by avbentem** [comment]
It seems that comments that were edited before the new dropdown was added a few days ago, only show a single entry with the same date as the original comment. I'm now curious if GitHub really did not keep track of changes before starting to show these dates...

**Comment by GolfingSuccess** [comment]
@avbentem I'm not sure if it actually keeps track of changes or just dates.

**Comment by aensley** [comment]
_Replying to https://github.com/dear-github/dear-github/issues/129#issuecomment-343656883 by @avbentem:_

I'm sure they haven't kept the old contents, and they still aren't. I'm glad they've at least added a full history of authors and edit times going forward. One step away from full edit history. I remain hopeful.

**Comment by hoganlong** [comment]
@aensley this is a version control stystem, it is easier to keep the history than to just keep the times.  Why not keep the history?

**Comment by aensley** [comment]
_Replying to https://github.com/dear-github/dear-github/issues/129#issuecomment-345403392 by @hoganlong:_

I have no idea. I'm not GitHub. I find it humorous that my last comment is down-voted; as if I'm the one making the decision. All of these comments are my observations and expectations.

I think the last addition is awesome, and I still hope for full edit history. :+1:

**Comment by borekb** [comment]
By the way, GitHub support can already send you older versions of your issues and PR comments, it's just not in the UI yet.

**Comment by advanceddeveloper** [comment]
I am not a fan of this.

A user may accidentally post some sensitive or private information when reporting an issue. Making edit history permanently visible would allow anyone to access such information. The current implementation seems reasonable to me: if a user who made an edit wants to reveal that he had made a mistake, he can just put <s>a stroke line</s> over the previous sentence and add the correct one. But, if someone fixed a typo, I don't think it is worth storing in the history. Or if someone edited out private information.

Another idea is to allow anyone to access edit history, but also allow the author of the issue (or commit) to permanently redact some edits.

**Comment by gthackra** [comment]
you could always use a webhook to write out edits to a secondary db for storage/review etc. Webhooks are configurable to trigger on an issue comment edit and contain enough information to track issue comment edit history.

**Comment by aensley** [comment]
_Replying to https://github.com/dear-github/dear-github/issues/129#issuecomment-388794344 by @advanceddeveloper:_

This is a common objection that has been addressed many times in this thread. Please see:

https://github.com/dear-github/dear-github/issues/129#issuecomment-224027073

https://github.com/dear-github/dear-github/issues/129#issuecomment-302120674

https://github.com/dear-github/dear-github/issues/129#issuecomment-333889699

https://github.com/dear-github/dear-github/issues/129#issuecomment-334267824

**Comment by darkred** [comment]
Wow, GitHub has finally implemented this !!
https://twitter.com/GHchangelog/status/1000094809942507521 i.e. [Tracking changes in a comment](https://blog.github.com/changelog/2018-05-24-comment-edit-history/)

**Comment by hoganlong** [comment]
@advanceddeveloper note, they added the ability to delete into the UI too -- seems to address your concern.

**Comment by advanceddeveloper** [comment]
@aensley 

I hadn't really read all comments in this thread, I just posted my opinion, so I am not surprised that others had already said the same thing before.

@hoganlong 

I'm happy to hear that.

**Comment by borekb** [comment]
Well, it's been rolled back for now (confirmed by their support; they probably found some issue in it) but it will definitely come back. 🎉

**Comment by idgserpro** [comment]
Will comments edited before this implementation have the edit history diff or just edited comments _from now on_?

**Comment by idgserpro** [comment]
This makes sense. Imagine having edited a comment 2 years ago that you even don't remember which one was to remove sensitive information and trying to recover which one is to fix it...

Thanks.

**Comment by GolfingSuccess** [comment]
Another thing I'd like to have as a feature is, instead of being able to "delete", being able to "hide/unhide", and, in that way, revisions of comments prior to the announcement of public comment diffs would be hidden by default.

**Comment by jbruchon** [comment]
I'd like the ability to turn this feature off.

**Comment by dev-101** [comment]
I guess I am actually alone in this one, seeing how many of you requested this :)

I'm completely annoyed with this 'feature'. At least, GutHub should introduce 60-120 seconds grace period from initial posting to allow for minor grammar or typo fixes. It is really wasting resources to record 1 letter changes and typos imho, unless the post is edited much later or next day etc.

**Comment by avbentem** [comment]
> introduce 60-120 seconds grace period from initial posting

Or even much longer, unless maybe someone else meanwhile responded. 

Adding a grace period  would also require GitHub to postpone sending the email alerts a bit though, to ensure those reflect the most recent content. The alert I got included the first revision; no alert about the edit though.

**Comment by mesqueeb** [comment]
I want to vote +1 on 120 sec grace period + postponing the initial email of the comment until after these 120 sec.

**Comment by benbucksch** [comment]
-1   Personally, I expect a notification right away (less than 10 seconds until it's in my inbox) and consider a 2 minute email notification delay to be a bug. (I say that as someone who very often edits his comments.)

**Comment by Pacerier** [comment]
@FichteFoll re "*Discourse handles editing really well*".

Since it's written by SE's author who happens to be no web newb (unlike GH's founders!). Although batching updates (in the db) is stupid.

**Comment by Pacerier** [comment]
@aensley re "*no **expectations** of being able to delete it, even **if** that feature is available*".

Lol? That's not how the Internet works (unless the page is "high profile").

**Comment by Pacerier** [comment]
@Wildeast re "*right to delete old version should be restrictable to the repo owner*".

And for legal (?) conformity reasons the repo owner should not be allowed deleting of stuff he didn't write?

**Comment by Pacerier** [comment]
@dev-101 re "*completely annoyed*" "*minor grammar or typo fixes*".

You can don't edit it. and no one will be annoyed. But don't edit while wanting people to not know.

re "*minor*"; Typical for-you-but-not-me.

**Comment by jbruchon** [comment]
Hi! This is 7 years old. Maybe stop generating annoying email notifications for me over something that's a dead discussion? Thanks!

**Comment by luxurystorecc1** [comment]
Philadelphia Grit: Navigating the City of Brotherly Love with Elite Performance Gear
Philadelphia is a city defined by its unwavering passion for sports and its deep-rooted history. From the bustling Italian Market to the modern arenas of South Philly, residents demand gear that reflects the city's tough, resilient spirit. For sneaker enthusiasts in the 215, [authentic retro high-top sneakers in Philadelphia](https://www.google.com/search?q=https://www.airjordan-1.com/) are a cultural staple, merging classic hardwood performance with a modern urban edge.

The football culture in Philly is legendary. On game days, the city turns into a sea of green and silver, with fans prioritizing [official licensed professional league fan apparel](https://www.google.com/search?q=https://www.nflshopofficialonline-store.com/) to showcase their loyalty. For those who want a unique stadium look, [personalized gridiron jerseys with custom names](https://www.google.com/search?q=https://www.nflshopofficialonline-store.com/collections/custom-nfl-jersey) have become the gold standard for dedicated supporters who want to wear their pride personally.

Refined Performance for the Modern Professional
In the more affluent areas such as Rittenhouse Square, the preference for [luxury performance casual wear for men](https://www.peter-milla.com/) is evident. This technical apparel allows for a seamless transition from a high-stakes meeting to a relaxing afternoon at the golf club. Meanwhile, the younger generation continues to seek out [classic championship-inspired basketball footwear](https://www.google.com/search?q=https://www.jordan1s.com/) for its timeless appeal and reliable support on the city's public courts.

Comfort is essential for those navigating the cobblestone streets of Old City, leading to high interest in [breathable lightweight knit running shoes](https://www.yeezyboost-350.com/). On the pitch, Philly's international flair is reflected in the demand for [authentic international club soccer kits](https://www.soccerjerseysstore.com/). Competitive swimmers training at world-class Pennsylvania facilities rely on [professional racing swimwear and aquatic equipment](https://www.swim-outlet.com/) for peak efficiency.

Formal Style and Community Pride
For formal galas at the Philadelphia Museum of Art, many shoppers search for [fashionable plus size evening gowns for formal galas](https://www.cheapdressesformal.com/). To combat the humid East Coast summers, [structured team logo hats and caps](https://www.google.com/search?q=https://www.cheapnfl-jerseys.com/collections/headwear-nfl-headwear) are an essential outdoor accessory. Finally, as winter approaches, [official professional hockey league merchandise](https://www.custombaseball-jerseys.com/) remains a top-selling category for local fans.
